File: 闲来无事/Mob分享/Mob分享/ViewController.m
//
//  ViewController.m
//  Mob分享
//
//  Created by huangaengoln on 15/11/6.
//  Copyright © 2015年 huangaengoln. All rights reserved.
//

#import "ViewController.h"
#import <ShareSDK/ShareSDK.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    UIButton *shareBtn=[UIButton buttonWithType:UIButtonTypeCustom];
    shareBtn.frame=CGRectMake(100, 100, 100, 40);
    [shareBtn setTitle:@"分享" forState:UIControlStateNormal];
    shareBtn.backgroundColor=[UIColor yellowColor];
    [shareBtn addTarget:self action:@selector(shareClick:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:shareBtn];
    
}
-(void)shareClick:(UIButton *)sender {
    NSString *imagePath = [[NSBundle mainBundle] pathForResource:@"ShareSDK" ofType:@"png"];
    
    //1、构造分享内容
    id<ISSContent> publishContent = [ShareSDK content:@"要分享的内容"
                                       defaultContent:@"默认内容"
                                                image:nil
                                                title:@"ShareSDK"
                                                  url:@"http://www.mob.com"
                                          description:@"这是一条演示信息"
                                            mediaType:SSPublishContentMediaTypeNews];
    //1+创建弹出菜单容器
    id<ISSContainer> container = [ShareSDK container];
    /**
     *  iphone
     */
    [container setIPhoneContainerWithViewController:self];
    
    //2、弹出分享菜单
    [ShareSDK showShareActionSheet:container
                         shareList:nil
                           content:publishContent
                     statusBarTips:YES
                       authOptions:nil
                      shareOptions:nil
                            result:^(ShareType type, SSResponseState state, id<ISSPlatformShareInfo> statusInfo, id<ICMErrorInfo> error, BOOL end) {
                                
                                //可以根据回调提示用户。
                                if (state == SSResponseStateSuccess)
                                {
                                    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"分享成功"
                                                                                    message:nil
                                                                                   delegate:self
                                                                          cancelButtonTitle:@"OK"
                                                                          otherButtonTitles:nil, nil];
                                    [alert show];
                                }
                                else if (state == SSResponseStateFail)
                                {
                                    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"分享失败"
                                                                                    message:[NSString stringWithFormat:@"失败描述：%@",[error errorDescription]]
                                                                                   delegate:self
                                                                          cancelButtonTitle:@"OK"
                                                                          otherButtonTitles:nil, nil];
                                    [alert show];
                                }
                            }];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swifter_Second_Edition_100_development_of_the_necessary_Tip/Tip4多元组Tuple/Tip4多元组Tuple/ViewController.swift
//
//  ViewController.swift
//  Tip4多元组Tuple
//
//  Created by huangaengoln on 16/4/3.
//  Copyright © 2016年 huangaengoln. All rights reserved.
//

import UIKit


class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        // ==========例2
        //Objective-C中
        
        //
        let rect = CGRectMake(0, 0, 100, 100)
        let (small, large) = rect.divide(20, fromEdge: .MinXEdge)
        print("small : \(small) , large : \(large)")
        
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    // ========== 例1
    //以前的写法
    func swapMe<T>(inout a: T, inout b: T) {
        let temp = a
        a = b
        b = temp
    }
    //现在的写法
    func swapMe1<T>(inout a: T, inout b: T) {
        (a, b) = (b, a)
    }
    

}

extension CGRect {
    //...
    func divide(atDistance: CGFloat, fromEdge: CGRectEdge) -> (slice: CGRect, remainder: CGRect) {
        //function body
        var sliceTemp = CGRect()
        var remainderTemp = CGRect()
        CGRectDivide(self, &sliceTemp, &remainderTemp, atDistance, fromEdge)
        return (sliceTemp ,remainderTemp)
    }
    //...
}

